cxx-squid/src/main/java/org/sonar/cxx/CxxConfiguration.java

```diff
@@ -142,4 +198,37 @@ public void setMissingIncludeWarningsEnabled(boolean enabled){
   public boolean getMissingIncludeWarningsEnabled(){
     return this.missingIncludeWarningsEnabled;
   }
+  
+  public void setCompilationPropertiesWithBuildLog(List<File> reports, String fileFormat, String charsetName) {
+    
+    if(reports == null) {
+      return;
+    }
+    
+    for(File buildLog : reports) {
+      if (buildLog.exists()) {
+        LOG.debug("Parse build log  file '{}'", buildLog.getAbsolutePath());
+        if (fileFormat.equals("Visual C++")) {
+          cxxVCppParser.parseVCppLog(buildLog, charsetName);
+        }
+
+        LOG.debug("Parse build log OK: includes: '{}' defines: '{}'", uniqueIncludes.size(), uniqueDefines.size());
+      } else {
+        LOG.error("Compilation log not found: '{}'", buildLog.getAbsolutePath());
+      }    
+    }
+    
+    RaiseIssuesForNotFoundIncludes();
+  }
+
+  private void RaiseIssuesForNotFoundIncludes() {
```

Comment:
Havent finish this. The idea is to raise issues if a include path was used
that is invalid.

On Sun, Apr 12, 2015, 10:07 Günter Wirth <notifications@github.com> wrote:

> In cxx-squid/src/main/java/org/sonar/cxx/CxxConfiguration.java
> <https://github.com/wenns/sonar-cxx/pull/399#discussion_r28201274>:
>
> > +      if (buildLog.exists()) {
> > +        LOG.debug("Parse build log  file '{}'", buildLog.getAbsolutePath());
> > +        if (fileFormat.equals("Visual C++")) {
> > +          cxxVCppParser.parseVCppLog(buildLog, charsetName);
> > +        }
> > +
> > +        LOG.debug("Parse build log OK: includes: '{}' defines: '{}'", uniqueIncludes.size(), uniqueDefines.size());
> > +      } else {
> > +        LOG.error("Compilation log not found: '{}'", buildLog.getAbsolutePath());
> > +      }
> > +    }
> > +
> > +    RaiseIssuesForNotFoundIncludes();
> > +  }
> > +
> > +  private void RaiseIssuesForNotFoundIncludes() {
>
> What is the idea of this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wenns/sonar-cxx/pull/399/files#r28201274>.
>